internal/rpc/rpcserver/server.go

```diff
@@ -1814,6 +1814,47 @@ func (s *walletServer) RevokeTickets(ctx context.Context, req *pb.RevokeTicketsR
 	return &pb.RevokeTicketsResponse{}, nil
 }
 
+func (s *walletServer) RevokeTicket(ctx context.Context, req *pb.RevokeTicketRequest) (*pb.RevokeTicketResponse, error) {
+	var ticketHash *chainhash.Hash
+	var err error
+	if len(req.TicketHash) != 0 {
+		ticketHash, err = chainhash.NewHash(req.TicketHash)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "%v", err)
+		}
+	} else {
+		return nil, status.Errorf(codes.InvalidArgument, "tickethash is required to revoke ticket")
+	}
+
+	if len(req.Passphrase) > 0 {
```

Comment:
we no longer submit the passphrase to unlock the wallet for each separate request, because it does not work properly after we introduced per-account encryption. instead, this call should require that the wallet (or account) is already unlocked, and error otherwise.